src/lidar_mini.cpp

```diff
@@ -9,15 +8,29 @@
  * @license   MIT
  */
 
-LIDARmini::LIDARmini(hwlib::pin_in &RX) : RX(RX) {
+LIDARmini::LIDARmini() : uart(115200, UARTController::ONE, 1) {
 }
 
-char *LIDARmini::getDistanceInCm() {
-    UARTProtocol uart(RX, 1000);
+std::array<char, 7> LIDARmini::getSensorData() {
+    std::array<char, 7> bytes = {};
 
-    for (int i = 0; i < 8; ++i) {
-        bytes[i] = uart.getByte();
+    // The 100k value is chosen because if it is lower it does not work, we're not quite sure why.
+    // But we think it might be related to the BAUDrate
+    for (int j = 0; j < 100000; j++) {
+        if (uart.available() >= 9) { // 9 is the amound of bytes of the full data package
+            // Check if startbyte is available.., twice. receive() pops an element of a stack
+            if (uart.receive() == 0x59 && uart.receive() == 0x59) {
+                for (int i = 0; i < 7; i++) {  // Loop for 7 more times to print the remaining data package.
+                    bytes[i] = uart.receive(); // Fill the bytes array
+                }
+                return bytes;
```

Comment:
This return could be replaced by a break, since bytes is always returned at the bottom of the function.